### PR TITLE
Accept unprefixed tzdata archive paths and make installer extraction path-aware

### PR DESCRIPTION
**Pull request by jumpsmm7**
### Motivation

- Arch Linux ARM tzdata packages may include POSIX timezone members without a leading `./` in tar member names, which caused the validation or extraction flow to fail despite the payload being safe and usable.
- Preserve existing safety checks (no traversal, regular-file, valid `TZif` content, safe link handling) while making the updater and installer robust to both archive path styles and remaining BusyBox `/bin/sh` compatible.

### Description

- Relaxed the tzdata package validation in `tools/update-tzdata.sh` so the Python validation accepts normalized names that start with `usr/share/zoneinfo/posix/` (unprefixed) as well as `./usr/share/zoneinfo/posix/`, while keeping checks for regular files, `TZif` magic, unsafe links, and archive-root traversal intact.
- Added `recompress_xz_package()` helper usage (already present) and preserved BusyBox-compatible decompression behavior during download/recompression flows in `tools/update-tzdata.sh`.
- Made the installer extraction path-aware in `installer` by detecting whether the chosen `TZ_FILE` uses a `./` prefix and passing the matching directory argument to `tar` (`./usr/share/zoneinfo/posix` vs `usr/share/zoneinfo/posix`) so extraction succeeds regardless of the archive member naming style.
- Updated the regression test `tests/update-tzdata-package-info.sh` to expect the unprefixed POSIX timezone payload to be accepted and to assert that the installer supports unprefixed POSIX timezone paths, and regenerated `installer.md5sum` and `installer.sha256sum` sidecars.

### Testing

- Ran the repository quality and tzdata tests: `sh tests/code-quality-checks.sh` completed with all items passing and the tzdata package conversion regression reported OK.
- Verified syntax and checksums: `sh -n tools/update-tzdata.sh`, `sh -n tests/update-tzdata-package-info.sh`, and `sh -n installer` all returned success, and `sh tools/update-checksums.sh installer` updated checksum sidecars successfully.
- Executed the end-to-end package validation test: `sh tests/update-tzdata-package-info.sh` passed locally after the change.
- External network attempt to run `sh tools/update-tzdata.sh .` failed to fetch the GPG key due to the upstream Ubuntu keyserver returning `Server indicated a failure` (network/GPG keyserver error), so live mirror download was not completed in this environment (non-code failure).
- `shellcheck -s sh` was not available in the validation environment (reported as unavailable), so ShellCheck linting was not performed here.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a8e5174b5b883299f96a726c9f09052)

### CODE REVIEW

**Comment by jumpsmm7** [comment]
@codex review the complete PR for final merge readiness, including correctness, security, BusyBox ash compatibility, failure safety, interruption recovery, rollback behavior, permissions, checksums, CI coverage, missing (or broken/failing/non-up-to-date) regression tests, and edge cases

File: installer [hunk 1]
```diff
@@ -6552,8 +6552,18 @@ inst_AdGuardHome() {
...
 }
 
+filter_timezone_members() {
+	awk '!/\/$/ {timezone=$0; sub(/^\.\//, "", timezone); if (timezone ~ /^usr\/share\/zoneinfo\/posix\// && timezone !~ /(^|\/)\.\.(\/|$)/) print $0}'
+}
+
+format_timezone_menu() {
+	filter_timezone_members |
+		sort |
+		awk -v INDEX=0 '{timezone=$0; sub(/^\.\//, "", timezone); sub(/^usr\/share\/zoneinfo\/posix\//, "", timezone); ++INDEX; printf "  %d) %s\n", INDEX, timezone}'
```
[the remark this answers is not in the document]
Comment (jumpsmm7) [reply installer:6574]: @codex fix this issue

**Comment by jumpsmm7** [comment]
@coderabbitai full review
